bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/binding/generic/converter/ColorChannelHandler.java

```diff
@@ -77,32 +77,29 @@ public String toString(Command command) {
     public Optional<State> toState(String string) {
         State newState = UnDefType.UNDEF;
         if (string.equals(channelConfig.onValue)) {
-            if (state instanceof HSBType) {
-                newState = new HSBType(((HSBType) state).getHue(), ((HSBType) state).getSaturation(),
-                        PercentType.HUNDRED);
+            if (state instanceof HSBType hsbState) {
+                newState = new HSBType(hsbState.getHue(), hsbState.getSaturation(), PercentType.HUNDRED);
             } else {
                 newState = HSBType.WHITE;
             }
         } else if (string.equals(channelConfig.offValue)) {
-            if (state instanceof HSBType) {
-                newState = new HSBType(((HSBType) state).getHue(), ((HSBType) state).getSaturation(), PercentType.ZERO);
+            if (state instanceof HSBType hsbState) {
+                newState = new HSBType(hsbState.getHue(), hsbState.getSaturation(), PercentType.ZERO);
             } else {
                 newState = HSBType.BLACK;
             }
-        } else if (string.equals(channelConfig.increaseValue) && state instanceof HSBType) {
-            BigDecimal newBrightness = ((HSBType) state).getBrightness().toBigDecimal().add(channelConfig.step);
+        } else if (string.equals(channelConfig.increaseValue) && state instanceof HSBType hsbState) {
+            BigDecimal newBrightness = hsbState.getBrightness().toBigDecimal().add(channelConfig.step);
             if (HUNDRED.compareTo(newBrightness) < 0) {
                 newBrightness = HUNDRED;
             }
-            newState = new HSBType(((HSBType) state).getHue(), ((HSBType) state).getSaturation(),
-                    new PercentType(newBrightness));
-        } else if (string.equals(channelConfig.decreaseValue) && state instanceof HSBType) {
-            BigDecimal newBrightness = ((HSBType) state).getBrightness().toBigDecimal().subtract(channelConfig.step);
+            newState = new HSBType(hsbState.getHue(), hsbStategetSaturation(), new PercentType(newBrightness));
```

Comment:
Trying to beat the build in feedback. 😉
```suggestion
            newState = new HSBType(hsbState.getHue(), hsbState.getSaturation(), new PercentType(newBrightness));
```